### PR TITLE
docs: First pass with Rho docs

### PR DESCRIPTION
**Pull request by mmulji-ic**

### CODE REVIEW

File: docs/migration/cosmoshub-4-v8-Rho-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+# Cosmos Hub 4, v8-Rho Upgrade, Instructions
+
+This document describes the steps for validator and full node operators for the successful execution of the [v8-Rho Upgrade](https://github.com/cosmos/gaia/blob/main/docs/roadmap/cosmos-hub-roadmap-2.0.md#v8-rho-upgrade-expected-q1-2023), which contains the following main new features/improvement:
+
+* (gaia) Bump [ibc-go](https://github.com/cosmos/ibc-go) to [v3.4.0](https://github.com/cosmos/ibc-go/blob/v3.4.0/CHANGELOG.md) to fix a vulnerability in ICA. See [v3.4.0 CHANGELOG.md](https://github.com/cosmos/ibc-go/releases/tag/v3.4.0) and [v3.2.1 Release Notes](https://github.com/cosmos/ibc-go/releases/tag/v3.2.1) for details.
+* (gaia) Bump [cosmos-sdk](https://github.com/cosmos/cosmos-sdk) to [v0.45.12](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.45.12). See [CHANGELOG.md](https://github.com/cosmos/cosmos-sdk/blob/release/v0.45.x/CHANGELOG.md) for details.
+* (gaia) Bump [tendermint](https://github.com/tendermint/tendermint) to [0.34.24](https://github.com/tendermint/tendermint/tree/v0.34.24). See [CHANGELOG.md](https://github.com/tendermint/tendermint/blob/v0.34.24/CHANGELOG.md) for details.
```
Comment (yaruwangway) [inline docs/migration/cosmoshub-4-v8-Rho-upgrade.md]: please remove (gaia), (feat), (fix), (tests)

**Review by yaruwangway — APPROVED** [review]

File: docs/migration/cosmoshub-4-v8-Rho-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+### Backups
+
+Prior to the upgrade, validators are encouraged to take a full data snapshot. Snapshotting depends heavily on infrastructure, but generally this can be done by backing up the `.gaia` directory.
+If you use Cosmovisor to upgrade, by default, Cosmovisor will backup your data upon upgrade. See below [upgrade by cosmovisor](#method-ii-upgrade-using-cosmovisor-by-manually-preparing-the-gaia-v700-binary) section.
+
+It is critically important for validator operators to back-up the `.gaia/data/priv_validator_state.json` file after stopping the gaiad process. This file is updated every block as your validator participates in consensus rounds. It is a critical file needed to prevent double-signing, in case the upgrade fails and the previous chain needs to be restarted.
+
+### Testing
+
+For those validator and full node operators that are interested in ensuring preparedness for the impending upgrade, you can join in our v8-Rho public-testnet or run a v8-Rho local testnet.
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v8-Rho-upgrade.md]: The public testnet is at v9-Lambda now, but running a local testnet is always an option: https://github.com/cosmos/testnets/tree/master/local

File: docs/migration/cosmoshub-4-v8-Rho-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+### Current runtime, cosmoshub-4 (pre-v8-Rho upgrade) is running Gaia v7.0.x
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v8-Rho-upgrade.md]:
```suggestion
### Current runtime, cosmoshub-4 (pre-v8-Rho upgrade) is running Gaia v7.1.1
```

File: docs/migration/cosmoshub-4-v8-Rho-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+- Manual upgrade
+- Upgrade using [Cosmovisor](https://github.com/cosmos/cosmos-sdk/tree/master/cosmovisor)
+  - Either by manually preparing the new binary
+  - Or by using the auto-download functionality (this is not yet recommended)
+
+If you prefer to use Cosmovisor to upgrade, some preparation work is needed before upgrade.
+
+### Method I: manual upgrade
+
+Run Gaia v7.0.x till upgrade height, the node will panic:
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v8-Rho-upgrade.md]:
```suggestion
Run Gaia v7.1.1 till upgrade height, the node will panic:
```

File: docs/migration/cosmoshub-4-v8-Rho-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+
+#### Expected result
+
+When the upgrade block height is reached, you can find the following information in the log:
+
+```shell
+ERR UPGRADE "v8-Rho" NEEDED at height: 14099412: upgrade to v7-Theta and applying upgrade "v8-Rho" at height:14099412
+```
+
+Then the Cosmovisor will create `$GAIA_HOME/cosmovisor/upgrades/v7-Theta/bin` and download Gaia v7.0.0 binary to this folder according to links in the `--info` field of the upgrade proposal 97.
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v8-Rho-upgrade.md]:
```suggestion
Then the Cosmovisor will create `$GAIA_HOME/cosmovisor/upgrades/v8-Rho/bin` and download the Gaia v8.0.0 binary to this folder according to links in the `--info` field of the upgrade proposal 97.
```

File: docs/migration/cosmoshub-4-v8-Rho-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+Then you should get the following structure:
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── gaiad  #v7.1.1
+└── upgrades
+    └── v8-Rho
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v8-Rho-upgrade.md]: Check with SDK team if this needs to be lowercased (and for which Cosmovisor versions)

Comment (glnro) [reply docs/migration/cosmoshub-4-v8-Rho-upgrade.md]: Yeah I think it's similar to what's in the testnet docs. @tac0turtle

**Review by glnro — APPROVED** [review]

**Pull request merged**